src/routes/category.rs

```diff
@@ -0,0 +1,125 @@
+use crate::{
+    models::{
+        category::{Category, CreateCategory},
+        error::Error,
+        response::{Empty, Response},
+        UserRecordId,
+    },
+    utils::{category, session},
+    Result,
+};
+use rocket::{post, serde::json::Json, tokio::fs::remove_dir_all, State};
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+use surrealdb::{engine::remote::ws::Client, Surreal};
+
+#[derive(Serialize, Deserialize)]
+#[serde(crate = "rocket::serde")]
+pub struct CategoryData<'r> {
+    pub id: &'r str,
+    pub token: &'r str,
+
+    pub cat: CreateCategory,
```

Comment:
```suggestion
    pub data: CreateCategory,
```

Consider change this field to `data` instead of `cat`, it's odd.